src/test/java/org/snomed/snowstorm/core/data/services/TraceabilityLogServiceTest.java

```diff
@@ -644,6 +651,86 @@ void testRebaseWithChangeSupersededByOtherProject() throws Exception {
 		assertTrue(componentChange.isSuperseded());
 	}
 
+	@Test
+	void testExtensionUpgradingDependencyWithSameComponentEdited() throws ServiceException, InterruptedException {
+		Concept concept;
+		CodeSystem codeSystem;
+		String extMain = "MAIN/SNOMEDCT-TEST";
+		Relationship relationship;
+
+		// Create International Concepts
+		concept = conceptService.create(
+				new Concept().addFSN("Vehicle (vehicle)").addRelationship(ISA, Concepts.SNOMEDCT_ROOT),
+				MAIN
+		);
+		String vehicleId = concept.getConceptId();
+		concept = conceptService.create(
+				// Published with wrong modelling
+				new Concept().addFSN("Car (vehicle)").addRelationship(ISA, Concepts.SNOMEDCT_ROOT),
+				MAIN
+		);
+		String carId = concept.getConceptId();
+		String relationshipToBeInactivated = concept.getRelationships().iterator().next().getRelationshipId();
+
+		// Version International
+		codeSystem = codeSystemService.createCodeSystem(new CodeSystem("SNOMEDCT", MAIN));
+		codeSystemService.createVersion(codeSystem, 20220131, "20220131");
+
+		// Create Extension
+		codeSystemService.createCodeSystem(new CodeSystem("SNOMEDCT-TEST", extMain));
+		concept = conceptService.create(
+				new Concept().addFSN("Extension module (core metadata concept)").addRelationship(ISA, Concepts.MODULE),
+				extMain
+		);
+		String extModule = concept.getConceptId();
+		branchService.updateMetadata(extMain, Map.of(Config.DEFAULT_MODULE_ID_KEY, extModule));
+
+		// International fixes modelling (through classification)
+		concept = conceptService.find(carId, MAIN);
+		concept.getRelationships().iterator().next().setActive(false).updateEffectiveTime();
+		concept.getRelationships().add(new Relationship(ISA, vehicleId));
+		concept = conceptService.update(concept, MAIN);
+		List<String> intRelationships = getRelationshipIds(concept);
+
+		// Version International
+		codeSystem = codeSystemService.find("SNOMEDCT");
+		codeSystemService.createVersion(codeSystem, 20220731, "20220731");
+
+		// Extension fixes modelling (through classification)
+		concept = conceptService.find(carId, extMain);
+		concept.getRelationships().iterator().next().setActive(false).updateEffectiveTime();
+		concept.getRelationships().add(new Relationship(ISA, vehicleId));
+		concept = conceptService.update(concept, extMain);
+		List<String> extRelationships = getRelationshipIds(concept);
+
+		// Extension upgrades to International's July content
+		codeSystem = codeSystemService.find("SNOMEDCT-TEST");
+		codeSystemUpgradeService.upgrade(codeSystem, 20220731, false);
+
+		// Extension ends up with International version of inactivated Relationship post upgrade
+		relationship = relationshipService.findRelationship(extMain, relationshipToBeInactivated);
+		assertTrue(relationship.isReleased());
+		assertEquals(Concepts.CORE_MODULE, relationship.getModuleId());
+		assertEquals(20220731, relationship.getReleasedEffectiveTime());
+
+		// Assert
```

Comment:
After upgrade the traceability activity should be "superseded" for extension, because the version from international release is chosen.

Reply:
Thanks @CoderMChu.

I have added this as an assertion.